waagent

```diff
@@ -2691,6 +2729,19 @@ def GetLineStartingWith(prefix, filepath):
             return line
     return None
 
+def ShellEscape(wordList):
+    """
+    Escape a list or tuple of strings for Unix Shell as words, using the
+    byte-literal single quote.
+    
+    The resulting string is safe for use with ``shell=True`` in ``subprocess``,
+    and in ``os.system``. ``assert shlex.split(ShellEscape(wordList)) == wordList``.
+    """
+    if not isinstance(wordList, (tuple, list)):
+        wordList = (wordList,)
+
+    return " ".join("'{0}'".format(s.replace("'","'\\''") for s in wordList))
```

Comment:
Sorry, this just isn't sufficient. Suppose one of the elements of wordlist was `foo\'bar`. This substitution turns that into `foo\\'bar` which will be interpreted very differently (and in a breaking way) by the shell.